backend/src/nodes/impl/onnx/session.py

```diff
@@ -21,7 +20,7 @@ def create_inference_session(
                 "TensorrtExecutionProvider",
                 {
                     "device_id": gpu_index,
-                    "trt_engine_cache_enable": should_tensorrt_cache,
+                    "trt_engine_cache_enable": tensorrt_cache_path is not None,
```

Comment:
you made the disabled case empty string, didn't you? How would this be None ever?

Reply:
See `SettingParser#get_cache_location`. That one maps the empty string to None for a cleaner API on the python side.